### PR TITLE
Change functions to use HTTP.jl package instead of deprecated Requests and HttpCommon.

### PR DESCRIPTION
**Pull request by fernandogelin**
## Types of changes

This PR implements the following changes:
_(Please tick any or all of the following that are applicable)_

* [ ] :sparkles: New feature (A non-breaking change which adds functionality).
* [ ] :bug: Bug fix (A non-breaking change, which fixes an issue).
* [ ] :boom: Breaking change (fix or feature that would cause existing functionality to change).
* [X] :exclamation: Remove deprecated libraries

## :clipboard: Additional detail

  - Request calls used deprecated libraries Requests.jl and HttpCommon.jl
  - New calls use HTTP.jl library.

## :ballot_box_with_check: Checklist

- [x] :art: The changes implemented is consistent with the [julia style guide](https://docs.julialang.org/en/stable/manual/style-guide/).
- [ ] :blue_book: I have updated and added relevant docstrings, in a manner consistent with the [documentation styleguide](https://docs.julialang.org/en/stable/manual/documentation/).
- [ ] :blue_book: I have added or updated relevant user and developer manuals/documentation in `docs/src/`.
- [x] :ok: There are unit tests that cover the code changes I have made.
- [x] :ok: The unit tests cover my code changes AND they pass.
- [ ] :pencil: I have added an entry to the `[UNRELEASED]` section of the manually curated `CHANGELOG.md` file for this repository.
- [x] :ok: All changes should be compatible with the latest stable version of Julia.
- [ ] :thought_balloon: I have commented liberally for any complex pieces of internal code.

### CODE REVIEW

File: src/eutils.jl [hunk 1]
```diff
@@ -130,7 +130,7 @@ datetype, reldate, mindate, maxdate.
 """
 function elink(ctx::Associative=empty_context(); params...)
     params = process_parameters(params, ctx)
-    return Requests.post(string(baseURL, "elink.fcgi"), data=params)
+    return HTTP.request("POST", string(baseURL, "elink.fcgi"), data=params)
 end
 
 """
```
Comment (mirestrepo) [inline src/eutils.jl]: I don't think that HTTP.post accepts a parameter input 'data'. I thought it expects `body` instead. Are we sure the tests are working? They may be responding with 200, but not containing the desired content in the response. Maybe we should improve tests a little bit to check the content more in depth?

File: test/eutils.jl [hunk 2]
```diff
@@ -4,22 +4,22 @@
...
-        @test isa(parsexml(res.data), EzXML.Document)
+        @test isa(parsexml(res.body), EzXML.Document)
     end
 
     @testset "esearch" begin
         res = esearch(db="pubmed", term="asthma")
         @test res.status == 200
         @test startswith(res.headers["Content-Type"], "text/xml")
-        @test isa(parsexml(res.data), EzXML.Document)
+        @test isa(parsexml(res.body), EzXML.Document)
```
Comment (mirestrepo) [inline test/eutils.jl]: So fo instance, here we could check the number of articles it returns. Also I suggest using a more complex 'term' -- issues may appear when escaping complex terms e.g. """(Asthma[MeSH Major Topic]) AND ("1/1/2018"[Date - Publication] : "3000"[Date - Publication])"""

File: REQUIRE [hunk 3]
```diff
@@ -2,5 +2,6 @@ julia 0.6
 EzXML 0.5
 JSON
 Requests
-Gumbo 0.4.0 
+Gumbo 0.4.0
 HttpCommon 0.3.0
```
Comment (mirestrepo) [inline REQUIRE]: We should be dropping use of HttpCommon. It is also deprecated

File: src/eutils.jl [hunk 4]
```diff
@@ -104,7 +105,8 @@ Parameters: db, id, query_key, WebEnv, retstart, retmax, retmode, version.
...
 function esummary(ctx::Associative=empty_context(); params...)
     params = process_parameters(params, ctx)
-    return Requests.post(string(baseURL, "esummary.fcgi"), data=params)
+    body = HTTP.escapeuri(params)
+    return HTTP.request("POST", string(baseURL, "esummary.fcgi?", body))
 end
```
Comment (mirestrepo) [inline src/eutils.jl]: Not sure if it makes a difference, but should the body be passed as a parameters instead of as part of the url?

**Comment by mirestrepo** [comment]
The decrease in coverage from umls is because credentials, which are environment variables in Travis, are not shared to Pull Requests

**Comment by fernandogelin** [comment]
@mirestrepo here's the link to Travis showing the tests are passing: https://travis-ci.org/fernandogelin/BioServices.jl

**Pull request merged**